src/upstream.c

```diff
@@ -389,3 +460,21 @@ void sendUpstreamMsgToServer(void **resp_bytes, size_t resp_size)
 	}
 
 }
+
+/*
+	Internal function to parse wrp src/dest
+*/
+static char *get_src_dest_from_sub_req(char *upstreamDest)
```

Comment:
Use wrp dest parsing function

Reply:
We cannot use this wrp function here as this function is only for dest parsing.
For cloud-status retrieve requests, in addition to dest parsing we need to do source parsing as well in the reverse flow. i.e. while sending crud response to libparodus clients.
As it is not advised to use strtok, currently I have modified existing get_src_dest_from_req() with strchr and avoid strtok.

Reply:
As per the review suggestion, modified wrp-c parsing function to support both source and dest parsing and submitted PR in wrp-c. It is under review.